winter/schema/type_hint.py

```diff
@@ -0,0 +1,188 @@
+
+
+# noinspection PyUnusedLocal
+@register(dict)
+def dict_resolver(hint_class) -> YASGTypeInfo:
+    return YASGTypeInfo(openapi.TYPE_OBJECT)
+
+
+# noinspection PyUnusedLocal
+@register(decimal.Decimal)
+def decimal_resolver(hint_class) -> YASGTypeInfo:
+    return YASGTypeInfo(TYPE_DECIMAL, openapi.FORMAT_DECIMAL)
+
+
+# noinspection PyUnusedLocal
+@register(uuid.UUID)
+def uuid_resolver(hint_class) -> YASGTypeInfo:
+    return YASGTypeInfo(openapi.TYPE_STRING, openapi.FORMAT_UUID)
+
+
+# noinspection PyUnusedLocal
+@register(datetime.datetime)
+def datetime_resolver(hint_class) -> YASGTypeInfo:
+    return YASGTypeInfo(openapi.TYPE_STRING, openapi.FORMAT_DATETIME)
+
+
+# noinspection PyUnusedLocal
+@register(datetime.date)
+def date_resolver(hint_class) -> YASGTypeInfo:
+    return YASGTypeInfo(openapi.TYPE_STRING, openapi.FORMAT_DATE)
+
+
+# noinspection PyUnusedLocal
+@register(list, tuple, collections.Iterable)
+def iterable_resolver(hint_class) -> YASGTypeInfo:
+    args = getattr(hint_class, '__args__', None)
+    child_class = args[0] if args else str
+    child_type_info = get_basic_type_info_from_hint(child_class)
+    return YASGTypeInfo(openapi.TYPE_ARRAY, child=child_type_info)
+
+
+@register(enum.IntEnum)
+def intenum_resolver(enum_class: typing.Type[enum.IntEnum]) -> YASGTypeInfo:
+    return enum_resolver(enum_class)
+
+
+@register(enum.Enum)
+def enum_resolver(enum_class: typing.Type[enum.Enum]) -> YASGTypeInfo:
+    enum_values = [entry.value for entry in enum_class]
+    # Try to infer type based on enum values
+    enum_value_types = {type(v) for v in enum_values}
+
+    if len(enum_value_types) == 1:
+        type_info = get_basic_type_info_from_hint(enum_value_types.pop())
+    else:
+        type_info = YASGTypeInfo(openapi.TYPE_STRING)
+    type_info.enum = enum_values
+    return type_info
+
+
+@register(UnionType, checker=is_optional)
+def union_resolver(hint_class) -> YASGTypeInfo:
+    child_type = hint_class.__args__[0]
+    type_info = get_basic_type_info_from_hint(child_type)
+    type_info.nullable = True
+    return type_info
+
+
+@register(object, checker=dataclasses.is_dataclass)
+def dataclass_resolver(hint_class) -> YASGTypeInfo:
+    fields = dataclasses.fields(hint_class)
+
+    properties = {
+        field.name: get_basic_type_info_from_hint(field.type)
+        for field in fields
+    }
+    return YASGTypeInfo(type_=openapi.TYPE_OBJECT, properties=properties)
+
+
+def get_basic_type_info_from_hint(hint_class) -> YASGTypeInfo:
+    origin_type = get_origin_type(hint_class)
+
+    if inspect.isclass(origin_type):
+        types = origin_type.mro()
+    else:
+        types = type(origin_type).mro()
+
+    for type_ in types:
+        data = _resolvers_by_type.get(type_)
+        if data is None:
+            continue
+        resolver, checker = data
+
+        if checker is None or checker(hint_class):
+            return resolver(hint_class)
+
+    raise ValueError(f'Unknown type: {hint_class}')
```

Comment:
The function is designed to inspect what possible, so it's not an incorrect behavior to pass something unknown to the function. So I suggest to raise more specific exception instead of ValueError. Or just return None if no type info is inspected.